chainservice/builder.go

```diff
@@ -127,16 +128,16 @@ func (builder *Builder) createInstance() {
 	}
 }
 
-func (builder *Builder) buildFactory(forTest bool) error {
-	factory, err := builder.createFactory(forTest)
+func (builder *Builder) buildFactory(forTest bool, opts ...factory.StateDBOption) error {
```

Comment:
mixing option with boolean is misleading

Reply:
I use `SetStopHeight`